.github/workflows/main.yml

```diff
@@ -13,8 +13,87 @@ on:
+  windows-arm64:
+    runs-on: windows-latest
+    strategy:
+      matrix:
+        options: [
+          {config: Debug, package: NO, tools: OFF, docs: OFF},
+          {config: Release, package: YES, tools: ON, docs: ON}
+        ]
+    env:
+      # Skip downloading necessities for the load tests and source and golden
+      # images for the other tests since we can't test them on arm64 anyway.
+      GIT_LFS_SKIP_SMUDGE: 1
+
+      BUILD_DIR: "build/build-batch-vs2022"
+      CONFIGURATION: "${{matrix.options.config}}"
+      CMAKE_GEN: "Visual Studio 17 2022" # Only VS version on Windows Server 2022
+      CMAKE_TOOLSET: "ClangCL"
+
+      PACKAGE: ${{matrix.options.package}}
+      FEATURE_TOOLS: ${{matrix.options.tools}}
+      FEATURE_DOC: ${{matrix.options.docs}}
+
+      PLATFORM: "arm64"
+      SUPPORT_SSE: OFF
+
+      # No tests on arm64 since the host is x64 and can't run the built tests
+      FEATURE_JNI: OFF # Not sure of the state of JNI on arm64 windows
+      FEATURE_TESTS: OFF
+      FEATURE_LOADTESTS: OFF
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        # Fetch all history to make sure tags are
+        # included (used for version creation)
+        fetch-depth: 0
+
+    # Need doxygen if docs are supposed to be built
+    # This has been observed to fail due to Sourceforge host failure
+    # Can't control that so please manually re-run if we time out here
+    - name: Install Doxygen
+      if: matrix.options.docs == 'ON'
+      uses: crazy-max/ghaction-chocolatey@v2
+      with:
+        args: install doxygen.install
+
+    - name: Setup Secure File
+      if: matrix.options.package == 'YES'
+      run: iex ((New-Object Net.WebClient).DownloadString('https://raw.githubusercontent.com/appveyor/secure-file/master/install.ps1'))
+
+    - name: Install Signing Certificate
+      if: matrix.options.package == 'YES'
+      run: |
+        if ($env:CERT_PK_PASS) {
+          $cert_file = "the_khronos_group_inc.p12"
+          appveyor-tools\secure-file -decrypt "${cert_file}.enc" -secret $env:CERT_DC_SECRET -salt $env:CERT_DC_SALT
+          certutil -p $env:CERT_PK_PASS -importpfx My $cert_file NoExport
+          rm $cert_file
+          $env:WIN_CODE_SIGN_IDENTITY = 'The Khronos Group Inc'
+          $env:WIN_CS_CERT_SEARCH_MACHINE_STORE = 'ON'
+        } else {
+          $env:WINDOWS_CODE_SIGN_IDENTITY = ''
+        }
+
+    - name: Build arm64 Windows
+      run: powershell ./ci_scripts/build_win.ps1
+
+    - name: Get KTX version
+      if: matrix.options.package == 'YES'
+      id: ktx-version
+      run: |
+        $env:KTX_VERSION = Get-Content $env:BUILD_DIR/ktx.version
+        echo "::set-output name=ktx-version::$env:KTX_VERSION"
+
+    - name: Upload Artifact
+      if: matrix.options.package == 'YES'
+      uses: actions/upload-artifact@v2
+      with:
+        name: KTX-Software-${{ steps.ktx-version.outputs.ktx-version }}-Windows-arm64
+        path: ${{env.BUILD_DIR}}/KTX-Software-*.exe
```

Comment:
Or we could just have every build that lands automatically produce a pre-release and then from there an authorized user could promote a pre-release to a full release: https://github.com/marketplace/actions/automatic-releases

Reply:
I prefer to only make releases on tags. All the contributing builds from different CI services need to agree on the release name so all pieces end up in the same release and using the tag is the easiest way I know to guarantee that. If you already made it the way you described, please change it.

Reply:
It's currently set up so that it will always upload an artifact but whenever a release triggers a build, it will upload to that release's artifact page